schemas/isd.json

```diff
@@ -0,0 +1,236 @@
+               }
+            },
+            "required": ["coefficients", "equation_key"]
+       },
+       "starting_detector_sample": {
+          "description": "The sample on the detector where the data starts",
+          "type": "number"
+       },
+       "starting_detector_line": {
+          "description": "The line on the detector where the data starts",
+          "type": "number"
+       },
+       "focal2pixel_samples": {
+          "description": "Transformation x coefficients from focal plane (mm) to detector pixel coordinates",
+          "type": "array",
+          "items": {
+             "type": "number"
+           }
+       },
+       "focal2pixel_lines": {
+          "description": "Transformation y coefficients from focal plane (mm) to detector pixel coordinates",
+          "type": "array",
+          "items": {
+             "type": "number",
+             "minItems": 3,
+             "maxItems": 3
+           }
+       },
+       "sensor_location": {
+          "description": "(X, Y, Z) location of sensor in body-fixed frame.",
+          "type": "array",
+          "items": {
+             "type": "array",
+             "items" : {
+                "type": "number",
+                "minItems": 3,
+                "maxItems": 3
+            }
+         }
+      },
+      "sensor_velocity" : {
+         "description": "(v_x, v_y, v_z) velocity of sensor in body-fixed frame.",
+         "type": "array",
+         "items": {
+            "type": "array",
+            "items" : {
+               "type": "number",
+               "minItems": 3,
+               "maxItems": 3
+            }
+         }
+      },
+      "sun_position" : {
+         "description": "(x, y,z) of the sun in body-fixed frame.",
+         "type": "array",
+         "items": {
+            "type": "array",
+            "items" : {
+               "type": "number",
+               "minItems": 3,
+               "maxItems": 3
+            }
+         }
+      },
+      "sun_velocity" : {
+         "description": "(v_x, v_y, v_z) velocity of sun in body-fixed frame.",
+         "type": "array",
+         "items": {
+            "type": "array",
+            "items" : {
+               "type": "number",
+               "minItems": 3,
+               "maxItems": 3
+            }
+         }
+      },
+      "sensor_orientation" : {
+         "description": "Orientation of the sensor provided in quaternions",
+         "type": "array",
+         "items": {
+            "type": "array",
+            "description": "Each quaternion is of the form: (scalar, vector, vector, vector)" 
+            "items" : {
+               "type": "number",
+               "minItems": 4 ,
+               "maxItems": 4
+            }
+         }
+      },
+      "line_scan_rate": {
+         "description": "The line rates of the detector map with the associated start line and time",
+         "type": "array",
```

Comment:
To make this work for both the line scan and framing camera, it'd be great if this would take either an array of objects or a single value. That way the framing camera and constant rate line scanners can just pass a single value.

Reply:
:+1:

Reply:
:+1: I just implemented to package one of these and passing a scalar for say CTX would be ideal.